casper/src/main/resources/RevVault.rho

```diff
@@ -175,100 +219,139 @@ in {
-                authKeyValidCh, authKeyValidEitherCh,
-                parametersOkCh, parametersAndAuthOkCh,
-                split, eitherPurseCh, doDeposit
+          // `amount` is transferred from vault at `fromRevAddress` to vault at `toRevAddress`.
+          contract _transferTemplate(@fromRevAddress, purse, @toRevAddress, @amount, authKey, ret) = {
+            new
+              revAddressValid,
+              revAddressValidEither,
+              amountNonNegative,
+              authKeyValidCh,
+              authKeyValidEitherCh,
+              parametersOkCh,
+              parametersAndAuthOkCh,
+              split,
+              eitherPurseCh,
+              doDeposit
             in {
-              RevAddress!("validate", targetAddress, *revAddressValid) |
+              // `toRevAddress` is validated
+              RevAddress!("validate", toRevAddress, *revAddressValid) |
               @Either!("fromNillableError <-", *revAddressValid, *revAddressValidEither) |
-              @Either!("fromBoolean", amount > 0, "Amount must be positive value", *amountNonNegative) |
-              @AuthKey!("check", *authKey, (*_revVault, ownRevAddress), *authKeyValidCh) |
+              // `amount` is verified
+              @Either!("fromBoolean", amount >= 0, "Amount must be positive value", *amountNonNegative) |
+              // `authkey` is validated
+              @AuthKey!("check", *authKey, (*_revVault, fromRevAddress), *authKeyValidCh) |
               @Either!("fromBoolean <-", *authKeyValidCh, "Invalid AuthKey", *authKeyValidEitherCh) |
               @Either!("productR <-", *revAddressValidEither, *amountNonNegative, *parametersOkCh) |
               @Either!("productR <-", *parametersOkCh, *authKeyValidEitherCh, *parametersAndAuthOkCh) |
               @Either!("flatMap <-", *parametersAndAuthOkCh, *split, *eitherPurseCh) |
               for (_, retCh <- split) {
                 new amountPurseCh in {
+                  // `revAddress`, `amount`, `authKey` are all valid
                   purse!("split", amount, *amountPurseCh) |
                   @Either!("fromSingletonList <-", *amountPurseCh, "Insufficient funds", *retCh)
                 }
               } |
               @Either!("flatMap <-", *eitherPurseCh, *doDeposit, *ret) |
-              for (@p, retCh <- doDeposit) {
-                @{targetAddress | bundle0{*_revVault}}!("_deposit", p, *retCh)
+              for (@fromPurse, retCh <- doDeposit) {
+                @{toRevAddress | bundle0{*_revVault}}!("_deposit", fromPurse, *retCh)
               }
             }
           } |
-          // Deposits the entire balance in the fromPurse to the toPurse.
-          contract _depositTemplate(fromPurse, toPurse, retCh) = {
-            new amountCh, depositSuccessCh in {
-              toPurse!("getBalance", *amountCh) |
+          // Deposits the entire balance in `fromPurse` to `toPurse`
+          contract _depositTemplate(toPurse, fromPurse, retCh) = {
```

Comment:
@tgrospic I noticed you changed the order here before. I don't know why. Should I use your change? I think current naming is fine..